flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/expressions/ExpressionExtractionUtils.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.expressions;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.table.expressions.ApiExpressionUtils.unresolvedFieldRef;
+import static org.apache.flink.table.expressions.BuiltInFunctionDefinitions.PROCTIME;
+import static org.apache.flink.table.expressions.BuiltInFunctionDefinitions.ROWTIME;
+import static org.apache.flink.table.expressions.BuiltInFunctionDefinitions.WINDOW_END;
+import static org.apache.flink.table.expressions.BuiltInFunctionDefinitions.WINDOW_START;
+import static org.apache.flink.table.expressions.FunctionDefinition.Type.AGGREGATE_FUNCTION;
+
+/**
+ * Utilities for splitting {@link Expression} into subcategories, e.g. into projections, aggregations, window properties
```

Comment:
nit: I would still try to aim for 100 char line length. Otherwise we need scrolling, e.g. in github reviews.

Reply:
Did you mean 120? If so yes, I missed the 1 character limit, sorry :( 

If you meant 100, then I guess this should be first settled in the community, as I've heard contradicting opinions on that matter from other committers.

Reply:
For code I agree but for JavaDoc it doesn't harm to break earlier. You also do it at other locations. Even our license header is breaking earlier.